src/NUnitFramework/tests/Internal/EventListenerTextWriterTests.cs

```diff
@@ -270,64 +280,71 @@ public void TestWriteCharBufferSubstring()
-            Assert.That(TestResult.Outputs.Count, Is.EqualTo(2));
-            Assert.That(TestResult.Outputs[0], Is.EqualTo(expected));
-            Assert.That(TestResult.Outputs[1], Is.EqualTo(expected + NL));
+            Assert.That(ListenerResult.Outputs.Count, Is.EqualTo(2));
+            Assert.That(ListenerResult.Outputs[0], Is.EqualTo(expected));
+            Assert.That(ListenerResult.Outputs[1], Is.EqualTo(expected + NL));
         }
 
         [Test]
         public void TestWriteFloat()
         {
             float value = -5.0f;
 
-            TestWriter.Write(value);
-            TestWriter.WriteLine(value);
+            ListenerWriter.Write(value);
+            ListenerWriter.WriteLine(value);
 
             var expected = "-5";
-            Assert.That(TestResult.Outputs.Count, Is.EqualTo(2));
-            Assert.That(TestResult.Outputs[0], Is.EqualTo(expected));
-            Assert.That(TestResult.Outputs[1], Is.EqualTo(expected + NL));
+            Assert.That(ListenerResult.Outputs.Count, Is.EqualTo(2));
+            Assert.That(ListenerResult.Outputs[0], Is.EqualTo(expected));
+            Assert.That(ListenerResult.Outputs[1], Is.EqualTo(expected + NL));
         }
 
         [Test]
         public void TestWriteLine()
         {
-            TestWriter.WriteLine();
+            ListenerWriter.WriteLine();
 
             var expected = NL;
-            Assert.That(TestResult.Outputs.Count, Is.EqualTo(1));
-            Assert.That(TestResult.Outputs[0], Is.EqualTo(expected));
+            Assert.That(ListenerResult.Outputs.Count, Is.EqualTo(1));
+            Assert.That(ListenerResult.Outputs[0], Is.EqualTo(expected));
         }
 
         #region ITestListener implementation
 
-        private class TestListenerHelper : ITestListener
+        private class TestListenerIntercepter : ITestListener
         {
             public IList<string> Outputs { get; internal set; }
+            public ITestListener DefaultListener { get; internal set; }
 
-            public TestListenerHelper()
+            public TestListenerIntercepter(ITestListener defaultListener)
             {
+                DefaultListener = defaultListener;
                 Outputs = new List<string>();
             }
 
             void ITestListener.TestStarted(ITest test)
             {
-                // Intentionally empty
+                if (DefaultListener != null)
+                    DefaultListener.TestStarted(test);
             }
 
             void ITestListener.TestFinished(ITestResult result)
             {
-                // Intentionally empty
+                if (DefaultListener != null)
+                    DefaultListener.TestFinished(result);
             }
 
             void ITestListener.TestOutput(TestOutput output)
             {
                 Assert.IsNotNull(output);
                 Outputs.Add(output.Text);
+
+                if (DefaultListener != null)
+                    DefaultListener.TestOutput(output);
```

Comment:
Once this PR is merged, it will still be possible, but less likely. For example, if the user calls Write(char) explicitly with just \r.